src/Host.cpp

```diff
@@ -44,6 +44,141 @@
+        mIsRunning = true;
+        return true;
+    }
+
+    if (mIsRunning) {
+        // Nothing to do, already running
+        return false;
+    }
+
+    // Is stopped, so subtract elapsed time from current and set that to be the
+    // effective start time:
+    mEffectiveStartDateTime = QDateTime::currentDateTimeUtc().addMSecs(-mElapsedTime);
+    mIsRunning = true;
+    return true;
+}
+
+bool stopWatch::stop()
+{
+    if (!mIsInitialised) {
+        // Nothing to do, never started
+        return false;
+    }
+
+    if (!mIsRunning) {
+        // Nothing to do, already stopped
+        return false;
+    }
+
+    // Is running - so stop and note time:
+    mElapsedTime = mEffectiveStartDateTime.msecsTo(QDateTime::currentDateTimeUtc());
+    mIsRunning = false;
+    return true;
+}
+
+bool stopWatch::reset()
+{
+    if (!mIsInitialised) {
+        // Nothing to do, never started
+        return false;
+    }
+
+    if (!mIsRunning) {
+        // Not running, so reset elapsed time:
+        mElapsedTime = 0;
+        // And reset initialised flag:
+        mIsInitialised = false;
+        return true;
+    }
+
+    // Is running so reset effective start time - BUT THIS DOES NOT stop the
+    // stopwatch:
+    mEffectiveStartDateTime = QDateTime::currentDateTimeUtc();
+    return true;
+}
+
+void stopWatch::adjustMilliSeconds(const qint64 adjustment)
+{
+    if (!mIsInitialised) {
+        // We can initialise things in this case by setting the flag and falling
+        // through into the is not running situation - with the elapsed time
+        // being zero up to now we just have to add on the adjustment:
+        mIsInitialised = true;
+    }
+
+    if (!mIsRunning) {
+        // Not running so adjust stored elapsed time:
+        mElapsedTime += adjustment;
+    }
+
+    // Is running so adjust effective start time - to increase the effective
+    // elapsed time we must subtract the adjustment from the effect start time:
+    mEffectiveStartDateTime = mEffectiveStartDateTime.addMSecs(-adjustment);
+}
+
+qint64 stopWatch::getElapsedMilliSeconds() const
+{
+    if (!mIsInitialised) {
+        // Never started so no elapsed time:
+        return 0;
+    }
+
+    if (!mIsRunning) {
+        // Not running - so return elapsed time:
+        return mElapsedTime;
+    }
+
+    // Is running so calculate elapsed time:
+    return mEffectiveStartDateTime.msecsTo(QDateTime::currentDateTimeUtc());
+}
+
+QString stopWatch::getElapsedDayTimeString() const
+{
+    if (!mIsInitialised) {
+        return QStringLiteral("+:0:0:0:0:000");
+    }
+
+    qint64 elapsed = 0;
+    if (mIsRunning) {
+        elapsed = mEffectiveStartDateTime.msecsTo(QDateTime::currentDateTimeUtc());
+    } else {
+        elapsed = mElapsedTime;
+    }
+
+    bool isNegative = false;
+    if (elapsed < 0) {
+        isNegative = true;
+        elapsed *= -1;
+    }
+
+    qint64 days = elapsed / 86400000L;
```

Comment:
Well - since I don't really have the inclination to learn all about https://en.cppreference.com/w/cpp/chrono - I could add comments along the line of, e.g. for this one of `24 hours x 60 minutes x 60 seconds x 1000 milliseconds` . I just happen to remember that there are 86400 seconds in a day... :man_shrugging:

Reply:
```cpp

#include <chrono>
using namespace std::chrono_literals;

//...
qint64 days = elapsed / 24h;
```

:)

Reply:
It is not as simple as that... :frowning_face: 
![Screenshot_20191002_210046](https://user-images.githubusercontent.com/6163092/66077373-bc904e00-e557-11e9-9937-5a39180d512f.png)

It does not seem possible to divide an integer (actually a `long long`) by a `chrono::hours` (aka `duration<long, ratio<3600> >`) - so what is the solution to this?

Reply:
`qint64 days = std::chrono::seconds(elapsed) / 24h;`

Reply:
Can we please not have this soup of qint64, quint8, quint16 and just stick with `auto`? It'll make things simpler to understand, at not much of an expense as far as I understand.